File: 2149. Rearrange Array Elements by Sign/rearrangeArray.cpp
class Solution
{
public:
    vector<int> rearrangeArray(vector<int> &nums)
    {

        queue<int> pos;
        queue<int> neg;
        for (auto it : nums)
        {
            if (it < 0)
            {
                neg.push(it);
            }
            else
                pos.push(it);
        }

        vector<int> ans;
        while (pos.size() > 0)
        {
            ans.push_back(pos.front());
            ans.push_back(neg.front());
            pos.pop();
            neg.pop();
        }

        return ans;
    }
};

File: 2149. Rearrange Array Elements by Sign/rearrangeArray.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        
        int pos = 0;
        int neg = 1;
        int n = nums.length;
        int ans[]  = new int[n];
        for(int val : nums){
            if(val<0){
                ans[neg] = val;
                neg = neg + 2;
            } else { 
                ans[pos] = val;
                pos = pos + 2;
            }
        }

        return ans;
    }
}